controllers/bootstrapdata.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"encoding/json"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha2"
+	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/cloudinit"
+)
+
+// BootstrapDataGenerator is responsible for generating bootstrap data for a given config
+type BootstrapDataGenerator struct {
+	logr.Logger
+}
+
+// GenerateControlPlaneInit will generate the bootstrap data and set the fields necessary on the config object.
+func (u *BootstrapDataGenerator) GenerateControlPlaneInit(config *v1alpha2.KubeadmConfig) error {
+	u.Info("Generating user data for first control plane")
+
+	// TODO: use secrets for certs, plug into kubeadm
+	// TODO: sub in things in the cluster configuration like the cluster name and other data we can extract from the cluster object
+	config.Spec.ClusterConfiguration.APIVersion = "v1beta1"
+	config.Spec.ClusterConfiguration.Kind = "ClusterConfiguration"
+	clusterdata, err := json.Marshal(config.Spec.ClusterConfiguration)
```

Comment:
@chuckha 
There is a potential overlap between network settings in `CAPBK.v1alpha2.KubeadmConfig.ClusterConfiguration` and  network settings in `CAPI.v1alpha2.Cluster`. What is expected behavior here? (once I know the behavior, I can fix it also in a fallow PR)

Reply:
Same goes e.g for the `CAPBK.controlplaneEndpoint`/ `CAPI.APIEndpoints`

Reply:
yes, you're right. I'll open a ticket to address this later

Reply:
https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/issues/78